### ISSUE TITLE
Save and publish fails for content node due invalid property

### ISSUE THREAD

**Issue by bjarnef** [umbraco/Umbraco-CMS#17817]
### Which Umbraco version are you using? (Please write the *exact* version, example: 10.1.0)

13.5.2

### Bug summary

We have importing of data via Hangfire which create or update content node.

However save and publish fails due and invalid property.

```
using var cref = _umbracoContextFactory.EnsureUmbracoContext();

var node = _contentService.Create(model.NodeName ?? model.Title, parent.Key, Committee.ModelTypeAlias);

var saveResults = _contentService.SaveAndPublish(node);
if (!saveResults.Success)
{
    ThrowError(saveResults);
}

return saveResults.Content!;
```

![image](https://github.com/user-attachments/assets/ad0b8580-1305-45cf-9cd9-1ce16ab5f1e9)

Strange thing is that we don't set this property at all during import and the property isn't mandatory.

Top section is just a simple single block in Block List for a top/hero banner.

![image](https://github.com/user-attachments/assets/1e1d6e74-b118-48de-bb4d-02c71162133e)

![image](https://github.com/user-attachments/assets/6f09618d-9166-4db9-aced-250414c3a37f)

![image](https://github.com/user-attachments/assets/6973b890-3c5e-4b4a-b396-738a6001eb45)

The node has been created, but not publish and I can save & publish the node from backoffice as expected.

In this case we crease Committee content nodes. Wee have pretty much identical import of Event content nodes, which doesn't fail, but the event nodes doesn't use the "Top" composition.

I am pretty sure this worked previously when I originally worked on the import from an Excel sheet in Umbraco v13.4, but the Umbraco Cloud project has since then been auto-upgraded to Umbraco v13.5

I wonder if any changes in Umbraco v13.5 has an impact on this as if I remove minimum value on the "BL - Top" datatype instance and thus no longer a single block list the import run and completes successfully.

![image](https://github.com/user-attachments/assets/e137f6eb-7c93-42c5-8100-ccc235786442)

I have a feeling that it is because the single block is scaffolded, when content node is created within backoffice, but not when created through content service. When saving and publishing the node from backoffice I guess it may populate the data, so property is valid when saved server side.

### Steps to reproduce

-

**Comment by bjarnef** [comment]
@nielslyngsoe do you recall if there is anything scaffolded for single but when save & publish node from within backoffice vs save & publish through ContentService?

**Comment by nikolajlauridsen** [comment]
Hey @bjarnef, this seems more of a question, have you tried asking the friendly folks over at Discord? They may have experienced something similar. 

Otherwise, please provide some reproduction steps from a clean setup, otherwise, it's quite tricky for us to fix it, if there is a bug 😄

**Comment by bjarnef** [comment]
@nikolajlauridsen well, to me it seems like a bug :) 

You can easily reproduce it by adding a composition with a single block to a content type.
Where save and publish from content service fails.

It you then than minimum value on block to zero (no longer single block) or just remove composition from content type, the save and publish from content service is successful :)

**Comment by nikolajlauridsen** [comment]
I'm not sure I understand, if you have a block list property with a requirement of at least 1 block, then it should be invalid if no block is present, seems like it's working as intended to me?

**Comment by bjarnef** [comment]
> I'm not sure I understand, if you have a block list property with a requirement of at least 1 block, then it should be invalid if no block is present, seems like it's working as intended to me?

Yes, you can see it like that.
However in backoffice one doesn't need to touch or fill anything in this property (because it is being scaffolded on client side), but from content service the value in `null`, but it expect some default JSON here (for the single block).

The property could however has a "valid" null value if the property (single block) first was added after the content node was created/published.

I guess it may not be classified as a real bug - perhaps we just need to make it simpler to handle scaffold of block(s) in Block List/Grid:
https://docs.umbraco.com/umbraco-cms/13.latest/fundamentals/backoffice/property-editors/built-in-umbraco-property-editors/block-editor/block-grid-editor#creating-a-block-grid-programmatically

Mainly a single block (but may also be needed if minimum number of blocks are required).

So basically what it does here client-side:
https://github.com/umbraco/Umbraco-CMS/pull/18073/files#diff-f1c9f2582c62c0091862c056c64ada8a981f554e121c1b08fbbf9eab1ec91b80R869

**Comment by nikolajlauridsen** [comment]
Ah okay, I see what you mean now. 

From a backend perspective, I don't think this is a bug and don't think we should add something to try and "auto-scaffold" certain items in certain scenarios. We have had some "magic" things like this in the past, but we always end up running into issues with it, and in general, it just creates an unclear and inconsistent API in my opinion. 

However I think that it would be nice, and a good idea, to have some helpers to facilitate more easily creating content programmatically in general, but especially around block items, however, this is very much a feature request. I think this could be something that would be a great candidate to be community-driven, it could even be its own package. 

In regards to the "valid" null scenario, you can make all kinds of havoc when you start adding stuff to document types after nodes a published, since we have no way to create the values, and I don't think we should either. 

Given all this, I'll go ahead and close this as "won't fix", however, please do make a discussion around the scaffolding, I think that could be a cool feature 😄

**Comment by bjarnef** [comment]
@nikolajlauridsen I have submitted a discussion for this:
https://github.com/umbraco/Umbraco-CMS/discussions/18218
